eitprocessing/datahandling/mixins/equality.py

```diff
@@ -24,6 +26,10 @@ def __eq__(self, other: Self):
             if len(t1) != len(t2):
                 return False
             return all(Equivalence._array_safe_eq(a1, a2) for a1, a2 in zip(t1, t2, strict=True))
+
+        if isinstance(self, UserDict) and isinstance(other, UserDict):
+            return all(Equivalence.__eq__(self[key], other[key]) for key in self)
```

Comment:
Above comment alleviates the need to check that both are `UserDict` (because if only one is, the method would return `None`)

```suggestion
        if isinstance(self, UserDict):
            return all(Equivalence.__eq__(self[key], other[key]) for key in self)
```

Reply:
That is correct.

There was a flaw in the logic here. It assumed that other has all the keys self had, but didn't check. Also, it didn't check whether other had any keys self didn't have.

I updated in in 294e06bf839f9a328d107b45ca72341adb7ddf73 to check all keys in self and other. If one of them doesn't have the key, the method returns `False`.